torch/csrc/jit/codegen/cuda/lower_scalar_hoist.cpp

```diff
@@ -308,7 +337,19 @@ std::pair<int64_t, bool> findAllocPointFromDataDependency(
     Val* value) {
   int64_t pos = -1;
   for (auto i : c10::irange(exprs.size())) {
-    for (auto o : exprs[i]->outputs()) {
+    auto expr = exprs[i];
+    if (auto alloc = dynamic_cast<kir::Allocate*>(expr)) {
+      // Currently this branch is only to handle shared memory address. For
+      // shared memory address, we generate code like `toSmem(T7)`, this does
+      // not need the shared memory tensor to be computed, but it do require T7
+      // to be allocated.
+      auto buffer = alloc->buffer();
+      if (buffer == value ||
```

Comment:
When `buffer == value`, doesn't it mean we found a complete match and we should return immediately with `{i, true}` much as line 354?

Reply:
No, `buffer == value` here can also match scalar buffer, and if this is the case, we should wait until next expr, which computes `buffer`. But if buffer is a tensor, then yes, we can immediately return.